autosynth/multi.py

```diff
@@ -6,93 +6,90 @@
 import functools
 import importlib
 import os
+import requests
 import subprocess
 import sys
+import typing
 
 import jinja2
 import yaml
 
-from autosynth import executor, github
+from autosynth import executor, github, synth
 from autosynth.log import logger
 
 
-def run(args, *, cwd=None, check=True, env=os.environ):
-    try:
-        return executor.run(
-            args,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-            cwd=cwd,
-            check=check,
-            encoding="utf-8",
-            env=env,
-        )
-    except subprocess.CalledProcessError as exc:
-        logger.error(
-            f"Failed executing {' '.join((str(arg) for arg in args))}:\n\n"
-            "{exc.stdout}"
-        )
-        raise exc
-
-
-def synthesize_libraries(libraries, github_token, extra_args):
-    results = []
-    for library in libraries:
-        logger.info(f"Synthesizing {library['name']}.")
-
-        command = [sys.executable, "-m", "autosynth.synth"]
-
-        env = os.environ
-        env["GITHUB_TOKEN"] = github_token
-
-        library_args = [
-            "--repository",
-            library["repository"],
-            "--synth-path",
-            library.get("synth-path", ""),
-            "--branch-suffix",
-            library.get("branch-suffix", ""),
-            "--pr-title",
-            library.get("pr-title", ""),
-        ]
-
-        if library.get("metadata-path"):
-            library_args.extend(["--metadata-path", library.get("metadata-path")])
-
-        if library.get("deprecated-execution", False):
-            library_args.append("--deprecated-execution")
-
-        result = run(
-            command + library_args + library.get("args", []) + extra_args,
-            check=False,
-            env=env,
-        )
-
-        results.append({"config": library, "result": result})
-
-    return results
-
-
-def _format_result(result):
-    error = False
-    skipped = False
-    returncode = result["result"].returncode
-
-    if returncode != 0:
-        if returncode == 28:
-            skipped = True
-        else:
-            error = True
-
+def _execute(command: typing.List[str], env: typing.Any) -> typing.Tuple[int, bytes]:
```

Comment:
I'm not sure how to reference the [custom type for _ENV](https://github.com/python/typeshed/blob/master/stdlib/3/subprocess.pyi#L37).